src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -29,12 +28,28 @@ public TypeInfo(ISchema schema)
         }
 
         /// <summary>
-        /// Returns a list of ancestors of the current node.
+        /// Returns an ancestor of the current node.
         /// </summary>
-        /// <returns></returns>
-        public INode[] GetAncestors()
+        /// <param name="index">Index of the ancestor; 0 for the node itself, 1 for the direct ancestor and so on.</param>
+        public INode GetAncestor(int index)
```

Comment:
@Shane32 `stackalloc` only works for primitive types and structures. But everything turned out to be even easier.